### PR TITLE
feat(DLI): add a new resource to manage flink template

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccFlinkTemplate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccFlinkTemplate_basic -timeout 360m -parallel 4
=== RUN   TestAccFlinkTemplate_basic
=== PAUSE TestAccFlinkTemplate_basic
=== CONT  TestAccFlinkTemplate_basic
--- PASS: TestAccFlinkTemplate_basic (15.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       15.217s
```

### CODE REVIEW

File: huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildGetFlinkTemplateQueryParams(d *schema.ResourceData) string {
+	res := "&limit=100"
+	if v, ok := d.GetOk("name"); ok {
+		res = fmt.Sprintf("%s&name=%v", res, v)
+	}
+
+	if res != "" {
```
Comment (houpeng80) [inline huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go]: it is unnecessary to check res, it is not empty forever

Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go]: done

File: huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The name of the flink template.`,
+			},
+			"sql": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go:47]: why the param is optional?

File: huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The statement of the flink template.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```
Comment (Lance52259) [inline huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go:56]:
```suggestion
```

File: huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go:47]: It‘s legal in this API

**Pull request merged**